File: FrontEnd/src/app/features/reservations/classes/dtos/form/reservation-write-dto.ts
import { Guid } from 'guid-typescript'
// Custom
import { PassengerWriteDto } from './passenger-write-dto'

export interface ReservationWriteDto {

    reservationId: Guid
    customerId: number
    destinationId: number
    pickupPointId: number
    portId: number
    date: string
    refNo: string
    ticketNo: string
    email: string
    phones: string
    adults: number
    kids: number
    free: number
    remarks: string
    passengers: PassengerWriteDto[]
    driverId?: number
    shipId?: number

}

File: API/Dtos/ReservationWriteDto.cs
using System;
using System.Collections.Generic;

namespace API.Dtos {

    public class ReservationWriteDto {

        // PK
        public Guid ReservationId { get; set; }
        // Fields
        public string RefNo { get; set; }
        public string Email { get; set; }
        public string Remarks { get; set; }
        // Navigation
        public List<PassengerWriteDto> Passengers { get; set; }

    }
}